src/App.tsx

```diff
@@ -15,36 +15,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+const getPreparedGoods = (
+  goods: string[],
+  { sortField, reversed }: { sortField: SortField; reversed: boolean },
+) => {
+  const copyGoods = [...goods];
+
+  if (sortField !== SortField.default) {
+    copyGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.alphabet:
+          return good1.localeCompare(good2);
+        case SortField.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    copyGoods.reverse();
+  }
+
+  return copyGoods;
+};
+
 export const App: React.FC = () => {
+  const [fildSort, setFildSort] = useState(SortField.default);
+  const [isReverse, setIsReverse] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField: fildSort,
+    reversed: isReverse,
+  });
+
+  const resetSort = () => {
+    setFildSort(SortField.default);
+    setIsReverse(false);
+  };
+
+  const reversGoods = () => {
+    setIsReverse(!isReverse);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => {
+            setFildSort(SortField.alphabet);
+          }}
+          type="button"
+          className={`button is-info ${fildSort !== SortField.alphabet ? 'is-light' : ''} `}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => {
+            setFildSort(SortField.length);
+          }}
+          type="button"
+          className={`button is-success ${fildSort !== SortField.length ? 'is-light' : ''} `}
```

Comment:
here and in other cases use [classnames](https://www.npmjs.com/package/classnames) library for managing  dynamic classnames.
already exists in project dependencies.